js/common/modules/@arangodb/aql/explainer.js

```diff
@@ -257,20 +257,26 @@ function printRules(rules, stats) {
     }
     return { name: key, time: stats.rules[key] };
   });
+  // filter out everything that was reasonably fast
+  times = times.filter((item) => item.time >= 0.0002);
+
   times.sort(function(l, r) {
     // highest cost first
     return r.time - l.time;
   });
   // top few only
   times = times.slice(0, 5);
 
-  stringBuilder.appendLine(section('Optimization rules with highest execution times:'));
-  stringBuilder.appendLine(' ' + header('RuleName') + '   ' + pad(maxNameLength - 'RuleName'.length) + header('Duration [s]'));
-  times.forEach(function(rule) {
-    stringBuilder.appendLine(' ' + keyword(rule.name) + '   ' + pad(12 + maxNameLength - rule.name.length - rule.time.toFixed(5).length) + value(rule.time.toFixed(5)));
-  });
+  if (times.length > 0) {
```

Comment:
wouldn't this if fit above the filtering of time duration line 260?

Reply:
I don't think so, because the filtering can modify the size/length of the `times` array.

Reply:
I was thinking it would be a starting condition before the filtering, but you're right. You don't want to know that you have time entries before filtering, as, after filtering, you might end up with none. So agree with the `if` remaining where it is.